### DISCUSSION TITLE
Want to contribute in carbon's documentation under GSoC'23.

### DISCUSSION THREAD

**Discussion by H9660** [carbon-language/carbon-lang#2676] [General]
<h1>Hi internet!</h1>

This is Hussain Lohawala, a penultimate year student in Chandigarh University. I am currently studying computer science.
After browsing the list of the project ideas, I had my interest laid down on the project `Creating documentation for Carbon source files`.

The project details are:

<b>Difficulty level</b>: \_\_\_\_
<b>Skills required/preferred</b>: C++, HTML, MarkDown, language parsing
<b>Project size</b>: 350 hr

Generate documentation from Carbon source files. This might include:

- Producing HTML and/or MarkDown output
- Converting declarations for classes, functions, interfaces, and so on.
- Extracting marked comments
- Recognizing commenting conventions for parameters, returns, cross reference, markdown code snippets, and so on.
- Cross referencing declarations, definitions, and possibly uses
- Continuous integration for deployment

<h1>Why me?</h1>

- I am quite familiar with the concepts of git that are used heavily in various open source projects.
- I have been accepted as a contributor in <b>Hackoctoberfest</b> twice in 2022 and 2023.
- I am really good at editing text with markdown.
- I have made some really cool projects in HTML, CSS and javascript and am really good at writing documention and explaining my work.
- In my previous semester I had covered subjects like Systems Programming in which I thoroughly studied langauge parsing in langauge processing.
- I practice problem solving daily at Leetcode and have maintained a streak of 94 days.

Here are some of my work:
 - Estates: [estatery-nine.vercel.app](https://estatery-nine.vercel.app/)
 - Wordle: [wordlev1.vercel.app](https://wordlev1.vercel.app/)
 - Cheer's Cafe: [cheer-cafe.vercel.app](https://cheer-cafe.vercel.app/)

<h1> My commitments </h1>

I commit the following as a contributor in Carbon:

- I commit to devote my full time in this project as I don't have any holiday plans during this summer.
- I commit to keep any confidential information about carbon secretive and not revealing it to third parties.
 - I commit to produce work that is original free from any plagiarism.
 - I commit to be active in this project and not miss any deadlines be if for milestones or any other evaluation.
 - I commit to keep on contributing to this repository even after I had completed this project with Carbon.
 - I commit to be in contact with the mentors throught the GSoC official period.

 Regards

**Comment by H9660** [comment]
I want your guidance in the same, @jonmeow.

**Comment by H9660** [comment]
Here is my proposal.
[Doc_generation_proposal_for_gsoc.docx](https://github.com/carbon-language/carbon-lang/files/10959464/Doc_generation_proposal_for_gsoc.docx)

Please review it @jonmeow

**Comment by jonmeow** [comment]
Please note that per [GSoC's timelines](https://developers.google.com/open-source/gsoc/timeline), applications are not currently being accepted. When applications open up on March 20, I think you'll need to apply through GSoC's website ([here's Carbon's page](https://summerofcode.withgoogle.com/programs/2023/organizations/carbon-language)). You might also want to read [Carbon's contributor guidance](https://docs.google.com/document/d/1Ly3tD58IE57h_xsoZ8wsL8y4Mp89fg8J8qLN8sCPZ5k/preview?resourcekey=0-CLlGUVSc7DMhLPTABpmjRQ). Applications here won't be included.

**Reply by H9660** [reply]
@jonmeow I acknowledge your point. I just needed a review on my proposal from the mentors. I am not actually submitting my application.

**Reply by chandlerc** [reply]
> @jonmeow I acknowledge your point. I just needed a review on my proposal from the mentors. I am not actually submitting my application.

If you'd like early feedback, you can share your proposal as you've done. But there isn't a formal review process before it is submitted as a GSoC application. Ultimately, you'll need to submit the application and then we'll review those when they come in via the GSoC process.

**Reply by H9660** [reply]
Alright sir, I got it.

**Reply by H9660** [reply]
BTW how many projects are expected to be selected by Carbon this time?

**Reply by chandlerc** [reply]
> BTW how many projects are expected to be selected by Carbon this time?

There is a maximum number of spots, and I'm not sure exactly what it is or if it is still subject to change, but its a very small number. I think if we have good applications (especially, ones we have reasonably high confidence will be successful at their project), we'll try to accept the maximum number, but again, that's going to be very small. We're also still a pretty small community and so don't have a lot of mentors available. =]

**Reply by H9660** [reply]
Thanks for the information. I could'nt find the discord server associated with carbon. Can you help me out?

**Reply by KateGregory** [reply]
If you go to https://github.com/carbon-language/carbon-lang (the main page) and scroll right down to the bottom, you'll find the Join Us section which includes the Discord link along with other things you should read such as the code of conduct.

**Reply by H9660** [reply]
Ok sir. Thanks a lot.

**Reply by jonmeow** [reply]
@H9660 "sir" is a formal way to address a man, but not typically appropriate for addressing a woman. Please bear in mind that we have both women and men in our community, and avoid using gendered terminology when you don't know the right choice.

**Comment by H9660** [comment]
Alright I'll keep that in mind.

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
